### PR TITLE
HPCC-19601 Connect logMsgManager only when dali is available

### PR DESCRIPTION
**Pull request by rpastrana**
- Moves dali log manager connection from esp main to initdali section
- Moves the disconnectLogMsgManagerFromDali() call to detachESPFromDali()
- Adds synchronized check for dali attach state in WsDfu.DFUDeleteFiles()

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [x] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by rpastrana** [comment]
This PR addresses code review issues from https://github.com/hpcc-systems/HPCC-Platform/pull/11110 
If this PR is merged, https://github.com/hpcc-systems/HPCC-Platform/pull/11110  should be closed.

**Comment by rpastrana** [comment]
@RussWhitehead please review.

**Review by RussWhitehead — COMMENTED** [review]
A couple trivial comments

File: esp/services/ws_dfu/ws_dfuService.hpp [hunk 1]
```diff
@@ -225,6 +227,26 @@ class CWsDfuEx : public CWsDfu
     void getFilePartsOnClusters(IEspContext &context, const char* clusterReq, StringArray& clusters, IDistributedFile* df, IEspDFUFileDetail& FileDetails,
         offset_t& mn, offset_t& mx, offset_t& sum, offset_t& count);
     bool getQueryFile(const char *logicalName, const char *querySet, const char *queryID, IEspDFUFileDetail &fileDetails);
+    bool attachServiceToDali() override
+    {
+        synchronized block(m_detachMutex);
```
Comment (RussWhitehead) [inline esp/services/ws_dfu/ws_dfuService.hpp]: Consider using a more lightweight synchronization construct, such as atomic or spin lock

**Comment by rpastrana** [comment]
tested: confrgmgr startup, wsdfu deletefile -> log to audit, wsdfu deletefile in detach, stalled detach request followed by deletefile req

**Comment by RussWhitehead** [comment]
@rpastrana Looks good

**Comment by rpastrana** [comment]
@richardkchapman please merge

**Comment by richardkchapman** [comment]
@rpastrana Can you squash

**Comment by rpastrana** [comment]
@richardkchapman squashed and ready to merge

**Pull request merged**